src/eval.jl

```diff
@@ -20,6 +20,17 @@ function getmodule(data, pos)
   getthing("$main.$sub", getthing(main, Main))
 end
 
+function ends_with_semicolon(line)
+    match = rsearch(line, ';')
+    if match != 0
+        for c in line[(match+1):end]
+            isspace(c) || return c == '#'
+        end
+        return true
+    end
+    return false
+end
```

Comment:
Really the repl should handle linebreaks and stuff too, given that it supports multiline input.

We may not be able to accept this for the editor unless it works reliably in the vast majority of cases, since input is more varied there – matching the repl in the console is fine though.